api4/channel.go

```diff
@@ -1112,24 +1112,27 @@ func addChannelMember(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Check join permission if adding yourself, otherwise check manage permission
-	if channel.Type == model.CHANNEL_OPEN {
-		if member.UserId == c.App.Session.UserId {
-			if !c.App.SessionHasPermissionToChannel(c.App.Session, channel.Id, model.PERMISSION_JOIN_PUBLIC_CHANNELS) {
-				c.SetPermissionError(model.PERMISSION_JOIN_PUBLIC_CHANNELS)
-				return
-			}
-		} else {
-			if !c.App.SessionHasPermissionToChannel(c.App.Session, channel.Id, model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS) {
-				c.SetPermissionError(model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS)
-				return
+	// Bypass permissions check if you are already joined to the channel
```

Comment:
It's not immediately obvious to me why membership in a channel implies the ability to add arbitrary people to the channel. Why isn't this delegated to the permissions system to always check `model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS`?

Reply:
Ok, the comment is not clear enough, and the approach is not good enough. The idea is, if you try to add again a user that is already added, you don't need to check the permissions, because it will return the membership itself, without "reading" it. This is included here because if not, when you don't have the permission to "JOIN_PUBLIC_CHANNELS", you will use this same API in certain situations to get the membership (basically is something like, give my the membership of this channel joining to it, if needed).

I think that use cases only makes sense when you are working with your own user, and if not we can end with some leek of information. So, I going to change it. (Sorry I've being with the flu lately, and my code is noticing it.)

Reply:
Fixed